crates/turborepo-lib/src/opts.rs

```diff
@@ -37,6 +38,54 @@ pub struct Opts<'a> {
     pub scope_opts: ScopeOpts,
 }
 
+impl<'a> Opts<'a> {
+    pub fn synthesize_command(&self) -> String {
+        let mut cmd = format!("turbo run {} ", self.run_opts.tasks.join(" "));
+        let mut flags: Vec<String> = vec![];
```

Comment:
I'd consider pushing directly to the string versus creating a vector. It'll be a little more simple and won't create a bunch of unnecessary allocations

Reply:
I did try that initially, but it doesn't turn out simpler. There are a few cases where we need to check for existence of any items before doing a join so that the spacing works out properly. Using an `.extend` call lets us skip all those checks, at the cost of allocating. I also wouldn't be concerned with allocation here, it's most likely < 20 short strings.

Reply:
I'm confused, where do we need to check for existence for spacing? My implementation preemptively puts a space before any flag, so it adds a single space only if it's needed.

Reply:
Oh, I see. Yeah, you can do it if you don't use `join`. I can switch it over.